src/diffusers/models/lora.py

```diff
@@ -355,13 +355,27 @@ def forward(self, hidden_states: torch.Tensor, scale: float = 1.0) -> torch.Tens
         if self.lora_layer is None:
             # make sure to the functional Conv2D function as otherwise torch.compile's graph will break
             # see: https://github.com/huggingface/diffusers/pull/4315
+            if self.padding_mode != "zeros":
+                hidden_states = F.pad(hidden_states, self._reversed_padding_repeated_twice, mode=self.padding_mode)
+                self.padding = (0, 0)
             return F.conv2d(
                 hidden_states, self.weight, self.bias, self.stride, self.padding, self.dilation, self.groups
             )
         else:
-            original_outputs = F.conv2d(
-                hidden_states, self.weight, self.bias, self.stride, self.padding, self.dilation, self.groups
-            )
+            if self.padding_mode != "zeros":
+                original_outputs = F.conv2d(
+                    F.pad(hidden_states, self._reversed_padding_repeated_twice, mode=self.padding_mode),
+                    self.weight,
+                    self.bias,
+                    self.stride,
+                    (0, 0),
+                    self.dilation,
+                    self.groups,
+                )
+            else:
+                original_outputs = F.conv2d(
+                    hidden_states, self.weight, self.bias, self.stride, self.padding, self.dilation, self.groups
+                )
```

Comment:
```suggestion
            if self.padding_mode != "zeros":
                original_outputs = F.conv2d(
                    F.pad(hidden_states, self._reversed_padding_repeated_twice, mode=self.padding_mode),
                    self.weight,
                    self.bias,
                    self.stride,
                    (0, 0),
                    self.dilation,
                    self.groups,
                )
            else:
                original_outputs = F.conv2d(
                    hidden_states, self.weight, self.bias, self.stride, self.padding, self.dilation, self.groups
                )
```

Can we refactor this part in the same way?

Reply:
@patrickvonplaten Hi, thank you for your guidance. I have refactored the code, please review it.